### ISSUE TITLE
Crash Report: FunkinCamera doesn’t support Non-GL renderers 

### ISSUE THREAD

**Issue by TechnikTil** [FunkinCrew/Funkin#3760]
### Issue Checklist

- [X] I have properly named my issue
- [X] I have checked the Issues/Discussions pages to see if my issue has already been reported

### Platform

Itch.io (Downloadable Build) - Windows

### Browser

None

### Version

0.5.3

### Description

Me and my buddy ol pal @crushernotdrip decided to play FNF with integrated graphics and GPU disabled, just to see how laggy it would be.

First off, we start laughing our asses off after we see that the title screen looked like THIS
![image](https://github.com/user-attachments/assets/85876aeb-3401-4b3a-a724-7a5b38fdb29c)

After that, we waited for the state to transition to the main menu. IT TOOK 10 SECONDS. When it finally finished, the game crashed.
![image](https://github.com/user-attachments/assets/0068c42f-d2af-4438-848b-cfc436049b45)

This was all yesterday. Today, I decided to do a little research on why it crashed, and eventually figured out the game is crashing because context3D is null in the main OpenFL Stage. Did more research on OpenFL Stages, and figured out it was null because the window renderer wasn’t OpenGL or WebGL. (see [this](https://github.com/FunkinCrew/openfl/blob/develop/src/openfl/display/Stage.hx#L1198))

Sorry that this looks like a full on LA assignment 😭

### Steps to Reproduce

1. Turn off your intergrated graphics (and Graphics Card) in Device Manager
2. Open FNF

### Crash logs

don’t have it, update later

**Comment by Hundrec** [comment]
Here you go: https://github.com/FunkinCrew/Funkin/issues/3589

**Comment by TechnikTil** [comment]
> Oh hey, I recognize this issue from somewhere! I'll go fetch it

> Here you go: #3589

guess I suck at searching issues then
should I close this?

**Comment by Hundrec** [comment]
Nah, I like the way you described the issue. It provides enough valuable information to stay up

**Comment by TechnikTil** [comment]
alr

**Comment by TechnikTil** [comment]
> I'm strongly considering disabling Cairo and setting the game's minimum system requirements to require support for OpenGL. Any system that's weak enough to not even support OpenGL is pretty much guaranteed to have issues running Funkin', and requiring shaders etc. to be modified to work with Cairo is a big ask.
> 
> To my understanding, any decently modern computer has at least integrated graphics, which should provide the necessary support even without a dedicated GPU.
> 
> I'm fine with adding support for Vulkan/Metal or whatever on the Lime side if that ends up being necessary (and you'd probably see a decent performance improvement from that), but Cairo is definitely not capable of running Funkin' at a decent framerate from my understanding.

Should I close my PR to this issue then?

**Comment by TechnikTil** [comment]
Closed since Funkin' will only run if OpenGL is available.
